lib/translator/common.h

```diff
@@ -23,6 +28,22 @@ void DebugLog(const char* format, ...);
 
 void SetDebugCallback(void (*callback)(const char*));
 
+template <typename T>
+bool ReadValue(absl::Span<const uint8_t> data, T& out) {
+  static_assert(std::is_pod_v<T>, "Only supports POD types");
+  if (sizeof(T) > data.size()) return false;
+  memcpy(&out, data.data(), sizeof(T));
+  return true;
+}
+
+template <typename T>
+bool WriteValue(const T& data, absl::Span<uint8_t> out) {
+  static_assert(std::is_pod_v<T>, "Only supports POD types");
+  if (out.size() > sizeof(T)) return false;
```

Comment:
Isn't the goal of this function simply to write the type into the span? If so, we would have: 
memcpy(out.data(), &data, sizeof(T)); 
and 
if (sizeof(T) > data.size()) return false;

Reply:
oops, you're right. in this case we would want to ensure we write the entire T obj.